src/cephadm/cephadm

```diff
@@ -3529,27 +3532,145 @@ def command_list_networks():
 
 ##################################
 
-
 def command_ls():
     # type: () -> None
 
     ls = list_daemons(detail=not args.no_detail,
-                      legacy_dir=args.legacy_dir)
+                      legacy_dir=args.legacy_dir,
+                      target_fsid=args.fsid)
     print(json.dumps(ls, indent=4))
 
 
-def list_daemons(detail=True, legacy_dir=None):
-    # type: (bool, Optional[str]) -> List[Dict[str, str]]
-    host_version = None
-    ls = []
+def get_enabled_ceph_units(fsid: str) -> List[str]:
+    try:
+        units = os.listdir(f'/etc/systemd/system/ceph-{fsid}.target.wants')
```

Comment:
the current list_daemons takes 10s on a physical server - this version makes it <1s...we have to improve on dealing with scale.

The whole idea is to reduce the data gathering, so we can run it more often and have more current data for orch and UI flows. We already have to understand what systemd is doing since we're creating unit files directly, the the right place and format - so I don't see this as a big deal personally.

systemd isn't exactly new.

Reply:
I get the idea, and totally support speeding things up, and if this PR gives an order of magnitude (or more) improvement, then that's excellent!

My question above was meant to be specifically only around the benefit gained by switching from systemctl to os.listdir, vs. the other changes in this PR.